### PR TITLE
Remove UVM from PanzerAdaptersSTK (partially)

### PR DESCRIPTION
**Pull request by JacobDomagala**
Fixed tests:
- tExodusFaceBlock
- tExodusEdgeBlock
- PanzerAdaptersSTK_initial_condition_builder
- tSTK_IO

### CODE REVIEW

File: packages/panzer/adapters-stk/src/stk_interface/Panzer_STK_CustomMeshFactory.cpp [hunk 1]
```diff
@@ -326,50 +326,47 @@ namespace panzer_stk {
...
-             itr=elements.begin();itr!=elements.end();++itr) {
-        int q = mesh.elementGlobalId(*itr) - OffsetToGlobalElementIDs_;
-        for (int k=0;k<8;++k) {
-          int loc = q*8 + k;
-          charge_density_by_local_ids.push_back(ChargeDensity_[loc]);
-          electric_potential_by_local_ids.push_back(ElectricPotential_[loc]);
+
+      Kokkos::View<double**, Kokkos::HostSpace> charge_density_by_local_ids("charge_density_by_local_ids", n_elements, dim_1);
+      Kokkos::View<double**, Kokkos::HostSpace> electric_potential_by_local_ids("electric_potential_by_local_ids", n_elements, dim_1);
+
```
Comment (bathmatt) [inline packages/panzer/adapters-stk/src/stk_interface/Panzer_STK_CustomMeshFactory.cpp:348]: @rppawlo do you know if we need a layout here?  I don't think so but not 100%

File: packages/panzer/adapters-stk/test/stk_interface_test/tSTK_IO.cpp [hunk 2]
```diff
@@ -397,31 +397,42 @@ void assignBlock(FieldContainer & block,FieldContainer & vertices, double val)
...
-      }
-   }
+   Kokkos::parallel_for(
+       cellCnt, KOKKOS_LAMBDA(size_t cell) {
+          for (std::size_t node = 0; node < nodeCnt; node++) {
+             block(cell, node) = val;
+          }
+       }
+   );
 }
```
Comment (bathmatt) [inline packages/panzer/adapters-stk/test/stk_interface_test/tSTK_IO.cpp:410]: deep_copy???

**Review by bathmatt — APPROVED** [review]
let the tester run, don't merge

**Review by alanw0 — COMMENTED** [review]
We should have a discussion at some point, possibly. I'm not familiar exactly with what you're doing, but it seems like you might be copying stk field data into kokkos views in order to use the data on GPU device... I wonder if it would be convenient or useful for you to consider stk's ngp-mesh and ngp-field which allow you to manage that kind of thing using an API with modify-on-host/device and sync-to-host/device methods.

File: packages/panzer/adapters-stk/src/stk_interface/Panzer_STK_CustomMeshFactory.cpp (hunk 1 above)
Comment (rppawlo) [reply packages/panzer/adapters-stk/src/stk_interface/Panzer_STK_CustomMeshFactory.cpp:348]: I don't think so. The special layouts are only needed for FAD scalar types.

**Comment by bathmatt** [comment]
This fails to build on cuda for me, will look into it and maybe push a patch or just get the issue back

File: packages/panzer/disc-fe/src/Panzer_Workset_Builder_impl.hpp [hunk 3]
```diff
@@ -222,55 +222,64 @@ panzer::buildBCWorkset(const WorksetNeeds & needs,
...
-        }
+      Kokkos::parallel_for(
+          vertex_coordinates.extent(1), KOKKOS_LAMBDA(std::size_t vertex)
+          {
+            const auto extent = Teuchos::as<std::size_t>(vertex_coordinates.extent(2));
+
+            for (std::size_t dim = 0; dim < extent; ++dim)
+            {
+              coords_view(cell, vertex, dim) = vertex_coordinates(side.second[cell].first, vertex, dim);
+            }
```
Comment (bathmatt) [inline packages/panzer/disc-fe/src/Panzer_Workset_Builder_impl.hpp:265]: Line 264 doesn't work for me and I'm amazed it did for you, but I can't compile.  @JacobDomagala what version of nvcc are you using?  

Anyway you can't have std::pair or std::vector so side.second[] has to be replaced with some view structure or the copy needs to happen on host.  I think the latter is better but,

**Comment by rppawlo** [comment]
@alanw0 - yes - I have been planning to talk to you about a transition to STK-NGP. Maybe we can meet in the next week or two? I'll ping you outside of github.

File: packages/panzer/disc-fe/src/Panzer_Workset_Builder_impl.hpp [hunk 4]
```diff
@@ -222,55 +222,64 @@ panzer::buildBCWorkset(const WorksetNeeds & needs,
...
 
-    PHX::View<int*> cell_local_ids_k = PHX::View<int*>("Workset:cell_local_ids",worksets[side->first].cell_local_ids.size());
-    for(std::size_t i=0;i<worksets[side->first].cell_local_ids.size();i++)
-      cell_local_ids_k(i) = worksets[side->first].cell_local_ids[i];
-    worksets[side->first].cell_local_ids_k = cell_local_ids_k;
+    auto cell_local_ids_k = PHX::View<int*>("Workset:cell_local_ids", worksets[side.first].cell_local_ids.size());
+    Kokkos::parallel_for(
+        worksets[side.first].cell_local_ids.size(), KOKKOS_LAMBDA(std::size_t i) {
+          cell_local_ids_k(i) = worksets.at(side.first).cell_local_ids[i];
+        }
```
Comment (bathmatt) [inline packages/panzer/disc-fe/src/Panzer_Workset_Builder_impl.hpp:274]: Again here.

**Review by bathmatt — CHANGES_REQUESTED** [review]

File: packages/panzer/disc-fe/src/Panzer_Workset_Builder_impl.hpp (hunk 3 above)
Comment (JacobDomagala) [reply packages/panzer/disc-fe/src/Panzer_Workset_Builder_impl.hpp:265]: I'm using nvcc 11.4

**Review by bathmatt — COMMENTED** [review]
This is good to go, @rppawlo any comments?

**Review by bathmatt — APPROVED** [review]

**Review by rppawlo — APPROVED** [review]
Looks good to me! Thanks!

**Comment by rppawlo** [comment]
Just built with these changes. There are a ton of new warnings on cuda for calling a host function from device code. We might need another PR to address. Suspect it is pulling in objects like sideset_info and mesh_info. Might just need to pull the views out of these objects so lambda capture only pulls in device specific objects.

**Pull request merged**